homeassistant/components/elkm1/__init__.py

```diff
@@ -465,31 +477,31 @@ def should_poll(self) -> bool:
         return False
 
     @property
-    def extra_state_attributes(self):
+    def extra_state_attributes(self) -> dict[str, Any]:
```

Comment:
Can we make `Any` more specific?

Reply:
Not in a meaningful way. It covers wide number of attributes across a number of objects.

Reply:
>  It covers wide number of attributes across a number of objects.

Does that sound concerning? Attributes should be very limited and minimal.

Reply:
I'd be glad to take this on as a future improvement.

Reply:
There is quite a bit of legacy here.  Ideally some of these attributes should be their own sensors
<img width="395" alt="Screen Shot 2022-04-14 at 11 52 09" src="https://user-images.githubusercontent.com/663432/163482630-5180611c-78f6-48ab-bd05-d6e9d3d65a8f.png">

https://developers.home-assistant.io/docs/core/entity/?_highlight=extra_state_attributes
`Entities that generate a significant amount of state changes can quickly increase the size of the database when the extra_state_attributes also change frequently. Minimize the number of extra_state_attributes for these entities by removing non-critical attributes or creating additional sensor entities.`

I agree this it outside of scope for this PR though

Reply:
I'm willing to tackle this in a future PR. I'll likely contact you offline for a bit of collaboration.